src/Codegen/Generator.hack

```diff
@@ -349,26 +357,34 @@ final class Generator {
     private async function collectObjects<T as Field>(string $from_path): Awaitable<vec<GeneratableObjectType>> {
         $script = await HHAST\from_file_async(HHAST\File::fromPath($from_path));
 
+        $interfaces = dict[];
         $objects = vec[];
         $query_fields = vec[];
         foreach ($script->getDescendantsOfType(HHAST\ClassishDeclaration::class) as $class_decl) {
             if ($class_decl->hasAttribute()) {
                 $rc = new \ReflectionClass($class_decl->getName()->getText());
+
+                $graphql_interface = $rc->getAttributeClass(\Slack\GraphQL\GQLInterface::class);
+                if ($graphql_interface is nonnull) {
```

Comment:
Should we throw an error if the interface is actually a class? Or is it fine to do:

```hack
<<GQLInterface("Foo", "Foo")>>
abstract class Foo {
    <<Field("bar", "bar")>>
    public function bar(): string {
        return "bar";
    }
}
```

That seems generally useful - maybe we should require that GQLInterfaces be represented by Hack interfaces _or_ Hack abstract classes?

Reply:
Oops we have a comment race condition :) I didn't see @mwildehahn's comment before commenting...I'm also OK with making things more restricted if it makes it easier for now, but I think we'll *eventually* want to support all these cases. 

The selling point of this framework is roughly "you write your Hack code naturally/take existing Hack code, then slap some <<GraphQL\... annotations on it and it magically works". If we add too many restrictions like "actually you can't put this on an abstract class, you must refactor it into an interface first" people can get frustrated about those. Such restrictions aren't completely unaviodable, but I think we want to minimize them over time.

Reply:
> Yea, that makes sense. Let's keep it like this and we can see how it feels to use. It's hard to have a strong opinion when we haven't used this yet smile

100% agreed. It's best to not overthink anything, because everything will most likely be rewritten at least once, probably multiple times before the final version.